### PR TITLE
GR_102 validation

### PR DESCRIPTION
**Pull request by rshunim**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11524

## Description
Add the new GR102 validation.
(Checks whether a content item has to_version higher than another one it uses)
"Validate that source's toversion <= target's toversion."

### CODE REVIEW

**Comment by rshunim** [comment]
Just the basic code, still does not work.
Additionally need to add UTs

File: .changelog/4586.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Moved GR102 validation to the new validate format. This validation checks whether a content item has to_version higher than another one it uses. Additionally, removed the warnings when the source's to_version is higher than the general default version.
```
Comment (dorschw) [inline .changelog/4586.yml:2]: see comment on GR101 about splitting changelogs

**Review by dorschw — COMMENTED** [review]
add a unit test please. 
You may want to change the GR101 unit test, and check the error code in the result, instead of duplicating the GR101 unit test.

**Comment by rshunim** [comment]
As it is a different validation, I think it is better to separate the tests @dorschw

File: demisto_sdk/commands/validate/validators/GR_validators/GR102_is_using_invalid_to_version_valid.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+class IsUsingInvalidToVersionValidator(BaseValidator[ContentTypes], ABC):
+    error_code = "GR102"
+    description = "Validate that source's toversion <= target's toversion."
+    rationale = (
+        "Prevent issues where used objects are not available due to a version mismatch."
+    )
+    error_message = (
+        "Content item '{0}' whose to_version is '{1}' is using content items:"
+        " {2} whose to_version is lower than {3}"
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GR_validators/GR102_is_using_invalid_to_version_valid.py:89]:
```suggestion
        " {2} whose to_version is lower than {3}, making them incompatible"
```

File: demisto_sdk/commands/validate/validators/GR_validators/GR102_is_using_invalid_to_version_valid.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    XSIAMReport,
+    CaseField,
+    CaseLayout,
+    CaseLayoutRule,
+]
+
+
+class IsUsingInvalidToVersionValidator(BaseValidator[ContentTypes], ABC):
+    error_code = "GR102"
+    description = "Validate that source's toversion <= target's toversion."
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GR_validators/GR102_is_using_invalid_to_version_valid.py:83]: match to the format of GR107/GR108

**Review by dorschw — APPROVED** [review]

**Review by dorschw — APPROVED** [review]

**Pull request merged**